axelrod/player.py

```diff
@@ -98,6 +98,15 @@ def play(self, opponent, noise=0):
             s1, s2 = self._add_noise(noise, s1, s2)
         update_histories(self, opponent, s1, s2)
 
+    def reproduce(self):
```

Comment:
* Using copy in a strategy can lead to deep-copying loops, we had to alter MindReader when round_robin was initially changed to use copy
* deepcopy is very slow and unnecessary for this purpose, there's no reason to copy complex objects